### PR TITLE
Split N/E grid computation out of Tlonlat, create NElonlat subroutine.

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Split N/E grid computation out of Tlonlat, create NElonlat subroutine.
- [X] Developer(s): 
    apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    early tests show bit-for-bit, running full test suite now
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No 
- [X] Please provide any additional information or relevant details below:

Closes https://github.com/CICE-Consortium/CICE/issues/897

When TLON, TLAT, ANGLET are on the CICE grid, Tlonlat is NOT called.  This meant N and E grid info was never computed.  This would fail during history writing with invalid values in N and E grid arrays.  And it would also cause problem if the C-grid were run with this type of CICE grid.

There are no test grids that have TLON, TLAT, ANGLET on them, so this error was not found in standard test suites.  This was detected by users.

Also fixes some indentation and write statements.

### CODE REVIEW

**Comment by phil-blain** [comment]
OK so we still write NLON et al. to the history files, even when using the B grid. This is tangential to the bug being fixed but would still be worth it, in my opinion. 

I'll take a closer  look at the diff later.

**Review by dabail10 — APPROVED** [review]
Oh, I see the problem now. I was focused on the C-grid variables.

**Comment by apcraig** [comment]
I agree, writing the N, E grid info with the B-grid is a separate issue.  I understand the reasons for not writing that info with the B-grid.  But I sort of favor keeping it for now.  It's just a few 2d arrays, it provides some info that might still be useful with the B-grid.  And other variables could depend on those dimensions.  For instance, if we decide to start computing and/or writing data on the N, E locations in the future with the B-grid, we'll need/want that info.  I understand we can add it back later too.  I guess I'm willing to turn off the B-grid writing of N, E.  At the same time, should we turn off the computation of the N, E variables if it's a B-grid?

There is a separate idea hanging out which is to write a history file just for grid variables that could be controlled separately from the standard history files.  For instance, it could be written just once during a long run, reducing the amount of data on history files even further.  I favor focusing on this feature rather than fussing over the N, E grid variables on the B-grid right now.  See https://github.com/CICE-Consortium/CICE/issues/613.

If there is general agreement that we should not compute and/or write the N, E grid variables when running the B-grid, I'll implement that.  Feel free to push for that :)

**Comment by daveh150** [comment]
I favor to not compute/write unused N,E when using B-grid. 

RE not writing grid in history files:  I am hesitant to remove the lat/lon from the history files. I often will look at a history file in Panoply for quick analysis and plots. It is nice to have projected views to include coastline overlays. If the lat/lon is not in the files it would prevent a projected view in Panoply or similar plotting tools. Could this be controlled with namelist options?

**Comment by apcraig** [comment]
Thanks for your input @daveh150.  

Regarding the separate grid file, that would be controlled by namelist.  I imagine there might be two namelist, one to turn on writing to the history file, one to turn on writing to a separate grid file.  Each could be toggled separately to give you exactly the behavior preferred.  And the default would probably be to maintain the current situation.

**Review by eclare108213 — APPROVED** [review]

**Comment by apcraig** [comment]
See #897.  I am going to leave the N/E grid initialization and history writing as is.  We can revisit this at some point in the future.

**Comment by apcraig** [comment]
But I did create a gx3 netcdf grid/kmt file with the TLON, TLAT, and ANGLET variables on it and I will add that as a test case.

**Comment by apcraig** [comment]
I just added a gx3 netcdf grid/kmt test.  This is not bit-for-bit with the standard gx3 runs but is definitely just roundoff initially (as expected).  I also verified that all the NE grid variables are defined on the history file and reasonable now.  Prior to this PR, some of those variables were undefined.  This fixes the problem as far as I can tell.

**Comment by apcraig** [comment]
@dabail10, I propose we add the new gx3 netcdf grid and kmt files to zenodo as this is now part of the test suite.  Not sure the best way to do that, but let me know if I can help.  The two files (grid_gx3t.nc and kmt_gx3t.nc) are here

cheyenne:  /glade/p/cesm/pcwg_dev/CICE_data/grid/gx3/

**Comment by dabail10** [comment]
These have been added to Zenodo and the table updated.

File: cicecore/cicedyn/infrastructure/ice_grid.F90 [hunk 1]
```diff
@@ -2581,12 +2642,10 @@ subroutine Tlatlon
...
-!         if (nx_block > 5+2*nghost .and. ny_block > 5+2*nghost) then
-         write(nu_diag,*) 'min/max ULON:', y1*rad_to_deg, y2*rad_to_deg
-         write(nu_diag,*) 'min/max ULAT:', y3*rad_to_deg, y4*rad_to_deg
-!         endif
-         write(nu_diag,*) 'min/max TLON:', x1*rad_to_deg, x2*rad_to_deg
-         write(nu_diag,*) 'min/max TLAT:', x3*rad_to_deg, x4*rad_to_deg
+         write(nu_diag,*) subname,' min/max ULON:', y1*rad_to_deg, y2*rad_to_deg
+         write(nu_diag,*) subname,' min/max ULAT:', y3*rad_to_deg, y4*rad_to_deg
+         write(nu_diag,*) subname,' min/max TLON:', x1*rad_to_deg, x2*rad_to_deg
+         write(nu_diag,*) subname,' min/max TLAT:', x3*rad_to_deg, x4*rad_to_deg
```
Comment (phil-blain) [inline cicecore/cicedyn/infrastructure/ice_grid.F90:2648]: shouldn't these prints logically be in `Tlatlon`  instead of `NElatlon` ?

**Review by phil-blain — COMMENTED** [review]
I did not have time to look at the code last week, but took a look this morning and left a few comments.

I'm not sure it warrants a follow-up PR though...

**Pull request merged**